experimental/benchmarks/node-dc-eis/run_node-dc-eis.sh

```diff
@@ -0,0 +1,297 @@
+#!/bin/bash
+#shopt -s -o nounset
+
+start=`date +%s`
+#set locations so we don't end up with lots of hard coded bits throughout script
+RESOURCE_DIR=$1
+REL_DIR=$(dirname $0)
+ROOT_DIR=`cd "${REL_DIR}/.."; pwd`
+echo "ROOT_DIR=${ROOT_DIR}"
+SCRIPT_DIR=${ROOT_DIR}/node-dc-eis
+ACMEAIR_DIR=${RESOURCE_DIR}/node-dc-eis
+NODEDCEIS_DIR=${RESOURCE_DIR}/Node-DC-EIS-cluster
+MONGO_DIR=${SCRIPT_DIR}
+MONGODB_PORT=28000
+
+export DB_URL="mongodb://127.0.0.1:${MONGODB_PORT}/node-dc-eis-db"
+
+#these may need changing when we find out more about the machine we're running on
+NODE_AFFINITY="numactl --physcpubind=0,1"
+MONGO_AFFINITY="numactl --physcpubind=2,3"
+RUNSPEC_AFFINITY="numactl --physcpubind=2,3"
+
+function usage() {
+  echo "USAGE:"
+  echo "Currently this script will run Node-DC-EIS-Cluster and mongodb, with the affinities as follows:"
+  echo "Node : $(echo ${node_affinity})"
+  echo "Mongodb : $(echo ${mongo_affinity})"
+}
+
+function mandatory() {
+  if [ -z "${!1}" ]; then
+    echo "${1} not set"
+    usage
+    exit
+  fi
+}
+
+function optional() {
+  if [ -z "${!1}" ]; then
+    echo -n "${1} not set (ok)"
+    if [ -n "${2}" ]; then
+      echo -n ", default is: ${2}"
+      export ${1}="${2}"
+    fi
+    echo ""
+  fi
+}
+
+function remove(){
+  if [ -f $1 ]; then
+    rm $1
+  fi
+}
+
+function archive_files() {
+  # archive files
+  echo -e "\n##BEGIN $TEST_NAME Archiving $(date)\n"
+  mv $LOGDIR_TEMP/$LOGDIR_PREFIX $RESULTSDIR
+  echo -e "Perf logs stored in $RESULTSDIR/$LOGDIR_PREFIX"
+  echo -e "\nCleaning up"
+  rm -r $LOGDIR_TEMP
+  echo -e "\n## END $TEST_NAME Archiving $(date)\n"
+}
+
+function kill_bkg_processes()   # kill processes started in background
+{
+  echo "Killing due to : $@"    
+  $MONGODB_COMMAND "stop"
+  pkill node
+  pids=$(ps -eo pid,pgid | awk -v pid=$$ '$2==pid && $1!=pid {print $1}')  # get list of all child/grandchild pids - this doesnt seem to work on nodejs benchmark machine....
+  echo "Killing background processes"
+  echo $pids
+  kill -9 $OTHERPID_LIST $pids || true  # avoid failing if there is nothing to kill
+}
+
+function on_exit()
+{
+  echo "Caught kill"
+  kill_bkg_processes "caught kill"
+  #kill -9 $PID_LIST $OTHERPID_LIST
+  archive_files
+  exit 1
+}
+
+function timestamp()
+{
+  date +"%Y%m%d-%H%M%S"
+}
+
+trap on_exit SIGINT SIGQUIT SIGTERM
+
+# VARIABLE SECTION
+
+# define variables
+declare -rx SCRIPT=${0##*/}
+TEST_NAME=node-dc-eis
+echo -e "\n## TEST: $TEST_NAME ##\n"
+echo -e "## OPTIONS ##\n"
+
+optional RESULTSDIR ${ROOT_DIR}/results
+optional TIMEOUT 600
+RESULTSLOG=$TEST_NAME.log
+SUMLOG=score_summary.txt
+
+optional DRIVERHOST
+optional NODE_FILE server-cluster
+optional CLUSTER_MODE false
+optional PORT 9000
+optional DRIVERCMD ${RESOURCE_DIR}/Node-DC-EIS-client/runspec.py
```

Comment:
runspec.py is part of Node-DC-EIS code base. Once you clone/install the workload in appropriate directory (/home/ios/node-dc-eis) then these scripts should work fine.